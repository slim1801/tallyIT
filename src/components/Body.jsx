import React from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import numeral from "numeral";
import costSelector from "../selectors/costSelector";

import {
  onEntryDateTimeChanged,
  onExitDateTimeChanged
} from "../reducers/form";

import { Flex, FlexBox, VerticalAlignFlex } from "../common-styles";
import DateTimePicker from "./DateTimePicker";

const FormRow = styled(FlexBox)`
  padding: 15px;
`;

const FormContainer = styled(Flex)`
  position: relative;
  color: white;
`;

const LabelColumn = styled(VerticalAlignFlex)`
  width: 150px;
`;

export function Body(props) {
  return (
    <FormContainer>
      <FormRow>
        <LabelColumn>Entry Time</LabelColumn>
        <Flex>
          <DateTimePicker
            value={props.entryDateTime}
            onDateTimeChanged={props.onEntryDateTimeChanged}
          />
        </Flex>
      </FormRow>
      <FormRow>
        <LabelColumn>Exit Time</LabelColumn>
        <Flex>
          <DateTimePicker
            value={props.exitDateTime}
            onDateTimeChanged={props.onExitDateTimeChanged}
          />
        </Flex>
      </FormRow>
      <FormRow>
        <LabelColumn>Total</LabelColumn>
        <Flex className="__total">${numeral(props.cost).format("0,0")}</Flex>
      </FormRow>
    </FormContainer>
  );
}

function mapStateToProps(state) {
  const { entryDateTime, exitDateTime } = state.form;
  return { entryDateTime, exitDateTime, cost: costSelector(state) };
}

const mapDispatchToProps = {
  onEntryDateTimeChanged,
  onExitDateTimeChanged
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Body);
