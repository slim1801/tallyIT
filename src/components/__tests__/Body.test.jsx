import React from "react";
import { Body } from "../Body";

describe("Body", () => {
  test("smoke test", () => {
    const BodyWrapper = shallow(<Body />);
    expect(BodyWrapper).toMatchSnapshot();
  });

  test("cost is displayed correct", () => {
    const BodyWrapper = shallow(<Body cost={10000} />);
    expect(BodyWrapper.find(".__total").text()).toBe("$10,000");
  });
});
