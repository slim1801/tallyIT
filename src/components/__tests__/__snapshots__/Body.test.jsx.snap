// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body smoke test 1`] = `ShallowWrapper {}`;
